### PR TITLE
Update Golang dependencies for OTEL API/SDK (#923)

### LINKED ISSUE #923: Update Golang Dependencies
Accounting, Checkout, and Product Catalog Services need update from 1.10-1.11 to 1.16 per #916

### PR DESCRIPTION
**Pull request by anurag-rajawat**
# Changes

Fix #923 

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: src/accountingservice/go.mod [hunk 1]
```diff
@@ -5,12 +5,12 @@ go 1.19
 require (
 	github.com/Shopify/sarama v1.37.2
 	github.com/sirupsen/logrus v1.9.0
-	go.opentelemetry.io/contrib/instrumentation/github.com/Shopify/sarama/otelsarama v0.36.3
-	go.opentelemetry.io/otel v1.11.0
+	go.opentelemetry.io/contrib/instrumentation/github.com/Shopify/sarama/otelsarama v0.36.1
```
Comment (fatsheep9146) [inline src/accountingservice/go.mod:8]: why otelsarama package is downgrade?

File: src/checkoutservice/go.mod [hunk 2]
```diff
@@ -6,30 +6,28 @@ require (
 	github.com/Shopify/sarama v1.37.2
 	github.com/google/uuid v1.3.0
 	github.com/sirupsen/logrus v1.8.1
-	go.opentelemetry.io/contrib/instrumentation/github.com/Shopify/sarama/otelsarama v0.36.2
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.32.0
-	go.opentelemetry.io/contrib/instrumentation/runtime v0.36.1
-	go.opentelemetry.io/otel v1.10.0
-	go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc v0.32.1
+	go.opentelemetry.io/contrib/instrumentation/github.com/Shopify/sarama/otelsarama v0.36.1
```
Comment (fatsheep9146) [inline src/checkoutservice/go.mod:9]: same here

File: src/accountingservice/go.mod (hunk 1 above)
Comment (anurag-rajawat) [reply src/accountingservice/go.mod:8]: When I ran the command `go get -u go.opentelemetry.io/otel/`, Go automatically downgraded the otelsarama package.

File: src/checkoutservice/go.mod (hunk 2 above)
Comment (anurag-rajawat) [reply src/checkoutservice/go.mod:9]: same here.

**Pull request closed without merging**